Dropdown/DropdownList.js

```diff
@@ -0,0 +1,181 @@
+import kind from '@enact/core/kind';
+import EnactPropTypes from '@enact/core/internal/prop-types';
+import Spotlight from '@enact/spotlight';
+import SpotlightContainerDecorator from '@enact/spotlight/SpotlightContainerDecorator';
+import Group from '@enact/ui/Group';
+import PropTypes from 'prop-types';
+import compose from 'ramda/src/compose';
+import React from 'react';
+
+import Item from '../Item';
+import RadioItem from '../RadioItem';
+import Scroller from '../Scroller';
+import Skinnable from '../Skinnable';
+
+import css from './Dropdown.module.less';
+
+const ContainerDiv = SpotlightContainerDecorator({enterTo: 'last-focused'}, 'div');
+
+const DropdownListBase = kind({
+	name: 'DropdownListBase',
+
+	propTypes: {
+		/**
+		 * The selections for Dropdown
+		 *
+		 * @type {String[]|Array.<{key: (Number|String), children: (String|Component)}>}
+		 */
+		children: PropTypes.oneOfType([
+			PropTypes.arrayOf(PropTypes.string),
+			PropTypes.arrayOf(PropTypes.shape({
+				children: EnactPropTypes.renderable.isRequired,
+				key: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
+			}))
+		]),
+
+		/**
+		 * Placement of the Dropdown List.
+		 *
+		 * @type {String}
+		 */
+		direction: PropTypes.string,
+
+		/**
+		 * Called when an item is selected.
+		 *
+		 * @type {Function}
+		 */
+		onSelect: PropTypes.func,
+
+		/**
+		 * Callback function that will receive the scroller's scrollTo() method
+		 *
+		 * @type {Function}
+		 */
+		scrollTo: PropTypes.func,
+
+		/**
+		 * Index of the selected item.
+		 *
+		 * @type {Number}
+		 */
+		selected: PropTypes.number,
+
+		/**
+		 * The current skin for this component.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		skin: PropTypes.string,
+
+		/**
+		 * State of possible skin variants.
+		 *
+		 * Used to style the internal [Scroller]{@link agate/Scroller.Scroller}.
+		 *
+		 * @type {Object}
+		 */
+		skinVariants: PropTypes.object,
+
+		/**
+		 * The width of DropdownList.
+		 *
+		 * @type {('smallest'|'small'|'medium'|'large'|'x-large'|'huge')}
+		 */
+		width: PropTypes.oneOf(['smallest', 'small', 'medium', 'large', 'x-large', 'huge'])
+	},
+
+	defaultProps: {
+		direction: 'below center'
+	},
+
+	styles: {
+		css,
+		className: 'dropdownList'
+	},
+
+	computed: {
+		className: ({children, direction, width, styler}) => styler.append(direction.substr(0, direction.indexOf(' ')), width, {dropdownListWithScroller: children.length > 4}),
+		groupProps: ({skin}) => (skin === 'silicon') ?
+			{childComponent: RadioItem, itemProps: {size: 'small', className: css.dropDownListItem, css}, selectedProp: 'selected'} :
+			{childComponent: Item, itemProps: {css}, selectedProp: 'selected'}
+	},
+
+	render: ({children, groupProps, selected, skinVariants, scrollTo, onSelect, ...rest}) => {
+		delete rest.width;
+		delete rest.direction;
+		delete rest.skin;
+
+		return (
+			<ContainerDiv
+				{...rest}
+				spotlightRestrict="self-only"
+			>
+				<Scroller
```

Comment:
Why we didn't adopt VirtualList here?

Reply:
It would have required a lot of styling refactoring and I wanted to preseve the current structure of the Dropdown list, based on the UX requirements of Gallium and Agate

Reply:
Could you be more specific? Which UX requirements of Gallium and Agate block the change?

Reply:
Item in Gallium and Silicon has different heights. One requirement is that the entire item height is highlighted with a different background-color on focus. VirtualList has a static 'itemSize' by default. It might be calculated based on the skin, but I see that counter-productive.
Here are the UX requirements and a screenshot with dropdown with virtualList: https://zpl.io/bPkLp3k https://zpl.io/29nBwyA
![image](https://user-images.githubusercontent.com/63335068/108381502-8a48c480-7210-11eb-9a88-dccb29a3af63.png)
The focus bg-color is applied only on the item.

Also, with VirtualList, I could get the Scroller to activate, no matter how many items it had.

Let me know your opinion

Reply:
I'm not sure why the focus bg-color is applied like that. The given size seems mismatched. Also, the scroller should not be activated when the items are not overflowed by design. 🤔 Do you have any branches that I can test with?
Well, I know the height of the item is different from the skins. But we could get them from the GUI guide.
If the contents of the dropdown many, the time will take more when the dropdown opens if it's Scroller.
I hope we could sync this part with sandstone if it's not impossible.

Reply:
I created a branch where a VirtualList is used. Can you take a look? 

feature/PLAT-133738-VirtualList

Reply:
I changed the code based on yours and the style for Gallium and Silicon seems to work with my code.
Please refer to feature/PLAT-133738-list-mk branch. https://github.com/enactjs/agate/compare/feature/PLAT-133738-VirtualList...feature/PLAT-133738-list-mk?expand=1
The selected css didn't work for silicon, but I guess you could figure out what's missing. If you can't, I'll continue on tomorrow.

Reply:
I was able to fix the css for selected items and scroller.